tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+	case "pod":
+		pods, _ := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: label})
+		if len(pods.Items) == 1 {
+			return true
+		}
+	case "service":
+		services, _ := clientset.CoreV1().Services(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: label})
+		if len(services.Items) == 1 {
+			return true
+		}
+	default:
+		t.Logf("Unsupported resource type for presence check: %s", resourceType)
+		t.FailNow()
+	}
+	return false
+}
+
+func podWithLabelIsReady(t *testing.T, label string) bool {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	g(t).Eventually(func() bool {
+		return resourceWithLabelIsPresent(t, "pod", label)
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(gomega.BeTrue())
+	k8s.RunKubectl(t, kubectlOptions, "wait", "--for=condition=Ready", "pod", "-l", label, fmt.Sprintf("--timeout=%s", RETRY_TIMEOUT))
+	return true
+}
+
+func ExtractUsernamePassword(t *testing.T, secretName, ns string) credentials {
+	kubectlOptions := k8s.NewKubectlOptions("", "", ns)
+	secret := k8s.GetSecret(t, kubectlOptions, secretName)
+	username := secret.Data["username"]
+	password := secret.Data["password"]
+	creds := credentials{string(username), string(password)}
+	return creds
+}
+
+func runCassandraQueryAndGetOutput(t *testing.T, query string) string {
+	cqlCredentials := ExtractUsernamePassword(t, "k8ssandra-superuser", namespace)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	// Get reaper service
+	output, _ := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", "-it", "k8ssandra-dc1-default-sts-0", "--", "/opt/cassandra/bin/cqlsh", "--username", cqlCredentials.username, "--password", cqlCredentials.password, "-e", query)
+	return output
+}
+
+func WaitForPodWithLabelToBeReady(t *testing.T, label string) {
+	g(t).Eventually(func() bool {
+		return podWithLabelIsReady(t, label)
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(gomega.BeTrue())
+}
+
+func GetStargateService(t *testing.T) v1.Service {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	clientset, err := k8s.GetKubernetesClientFromOptionsE(t, kubectlOptions)
+	g(t).Expect(err).To(gomega.BeNil())
+
+	services, err := clientset.CoreV1().Services(namespace).List(context.Background(), metav1.ListOptions{})
+	for _, service := range services.Items {
+		if strings.HasSuffix(service.ObjectMeta.Name, "-stargate-service") {
+			return service
+		}
+	}
+
+	t.Log("Couldn't find Stargate service")
+	t.FailNow()
+	return v1.Service{}
+}
+
+func MinioServiceName(t *testing.T) string {
+	kubectlOptions := k8s.NewKubectlOptions("", "", "minio")
+	minioService, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "get", "services", "-l", "app=minio", "-o", "jsonpath='{.items[0].metadata.name}'")
+	g(t).Expect(err).To(gomega.BeNil())
+
+	log.Printf("Minio service: %s", minioService)
+	return minioService
+}
+
+func CheckResourceIsPresent(t *testing.T, resourceType, label string) {
```

Comment:
Should we consider adding `WithLabel` to the name of the externally accessible function to indicate that used as a filter?

Reply:
Yes, that sounds right.